sdks/java/io/fileschematransform/src/main/java/org/apache/beam/sdk/io/fileschematransform/FileWriteSchemaTransformConfiguration.java

```diff
@@ -0,0 +1,215 @@
+
+  /** Uses the given {@link org.apache.beam.sdk.io.ShardNameTemplate} for naming output files. */
+  @Nullable
+  public abstract String getShardNameTemplate();
+
+  /** Configures the filename suffix for written files. */
+  @Nullable
+  public abstract String getFilenameSuffix();
+
+  /** Configures extra details related to writing CSV formatted files. */
+  @Nullable
+  public abstract CsvConfiguration getCsvConfiguration();
+
+  /** Configures extra details related to writing Parquet formatted files. */
+  @Nullable
+  public abstract ParquetConfiguration getParquetConfiguration();
+
+  /** Configures extra details related to writing XML formatted files. */
+  @Nullable
+  public abstract XmlConfiguration getXmlConfiguration();
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+
+    /**
+     * The format of the file content. Used as {@link String} key lookup of {@link
+     * FileWriteSchemaTransformFormatProviders#loadProviders()}.
+     */
+    public abstract Builder setFormat(String value);
+
+    /** Specifies a common prefix to use for all generated filenames. */
+    public abstract Builder setFilenamePrefix(String value);
+
+    /**
+     * Specifies to compress all generated shard files by default, append the respective extension
+     * to the filename. See {@link org.apache.beam.sdk.io.Compression} for expected values, though
+     * stringified in all lowercase format.
+     */
+    public abstract Builder setCompression(String value);
+
+    /** The custom delimiter to be used in place of the default ones ('\r', '\n' or '\r\n'). */
+    public abstract Builder setDelimiter(String value);
+
+    /** The number of output shards produced; a value of 1 disables sharding. */
+    public abstract Builder setNumShards(Integer value);
+
+    /** Uses the given {@link org.apache.beam.sdk.io.ShardNameTemplate} for naming output files. */
+    public abstract Builder setShardNameTemplate(String value);
+
+    /** Configures the filename suffix for written files. */
+    public abstract Builder setFilenameSuffix(String value);
+
+    /** Configures extra details related to writing CSV formatted files. */
+    public abstract Builder setCsvConfiguration(CsvConfiguration value);
+
+    /** Configures extra details related to writing Parquet formatted files. */
+    public abstract Builder setParquetConfiguration(ParquetConfiguration value);
+
+    /** Configures extra details related to writing XML formatted files. */
+    public abstract Builder setXmlConfiguration(XmlConfiguration value);
+
+    public abstract FileWriteSchemaTransformConfiguration build();
+  }
+
+  /** Configures extra details related to writing CSV formatted files. */
+  @DefaultSchema(AutoValueSchema.class)
+  @AutoValue
+  public abstract static class CsvConfiguration {
+
+    /** Content written to the top of every sharded file prior to the header. */
+    @Nullable
+    public abstract String getPreamble();
```

Comment:
for my edification - is this a thing? I was not familiar with it : )

Reply:
It's the body of text that proceeds a header.  Below is an example of types of things I've encountered with systems that export in CSV format.  The "preamble" part of it is the body of text that proceeds the `column1, column2, column3` header.

```
Fake Lab, Inc
Date of Sample: 2022-12-05
DNA Sample: 0afc7711-ec24-4ca7-ab5b-f70f643ea771
Tissue Sample: 4370bf27-2058-4211-bc15-fcb8d8b5d74c
Subject ID: aa485d80-0472-42ae-bf1e-d4c61ecf8189
Clone ID: c048906f-0625-4283-80c4-b4a04c1c8a80

column1, column2, column3
1, 2, 3
4, 5, 6
7, 8, 9
```

Reply:
very cool thanks!